service/history/workflow_task_handler_callbacks.go

```diff
@@ -538,8 +538,10 @@ func (handler *workflowTaskHandlerCallbacksImpl) handleWorkflowTaskCompleted(
 		newMutableState             workflow.MutableState
 		responseMutations           []workflowTaskResponseMutation
 	)
-	// hasBufferedEvents indicates if there are any buffered events which should generate a new workflow task
-	hasBufferedEvents := ms.HasBufferedEvents()
+	updateRegistry := weContext.UpdateRegistry(ctx, nil)
+	// hasBufferedEvents indicates if there are any buffered events or admitted updates which should generate a new
+	// workflow task.
+	hasBufferedEvents := ms.HasBufferedEvents() || updateRegistry.HasOutgoingMessages(false)
```

Comment:
`hasBufferedEventsOrMessages`? Longer but better describes what is there. And I would rename field in `workflowTaskHandler` too.

Reply:
Right. I can't think of a better name than that. I've renamed them.